runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,250 @@
+		}
+		return nil, fmt.Errorf("no dimensions to summarize")
+	}
+
+	timeDimExpr := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	if e.metricsView.TimeDimension != "" {
+		for _, dim := range e.metricsView.Dimensions {
+			if dim.Name == e.metricsView.TimeDimension {
+				expr, err := e.olap.Dialect().MetricsViewDimensionExpression(dim)
+				if err == nil {
+					timeDimExpr = expr
+				}
+				break
+			}
+		}
+	}
+
+	// Use a recent time window with proper timestamp formatting
+	dimensionTimeRange := defaultTimeRange.Max.Add(-DefaultSampleTimeWindow)
+	whereClause := fmt.Sprintf("WHERE %s >= '%s'", timeDimExpr, dimensionTimeRange.Format(time.RFC3339))
+
+	escapedTableName := e.olap.Dialect().EscapeTable(e.metricsView.Database, e.metricsView.DatabaseSchema, e.metricsView.Table)
+	sql := fmt.Sprintf("SELECT %s FROM %s %s LIMIT %d",
+		strings.Join(selectClauses, ", "),
+		escapedTableName,
+		whereClause,
+		summaryQueryLimit,
+	)
+
+	rows, err := e.olap.Query(ctx, &drivers.Statement{
+		Query:            sql,
+		Priority:         e.priority,
+		ExecutionTimeout: defaultExecutionTimeout,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute dimension summary query: %w", err)
+	}
+	defer rows.Close()
+
+	if !rows.Next() {
+		return nil, fmt.Errorf("dimension summary query returned no results")
+	}
+
+	values := make([]interface{}, len(dimensions)*statsPerDimension)
+
+	// Create scan targets
+	for i := 0; i < len(values); i++ {
+		values[i] = new(interface{})
+	}
+
+	if err := rows.Scan(values...); err != nil {
+		return nil, fmt.Errorf("failed to scan dimension summary results: %w", err)
+	}
+
+	// Build the result summaries
+	summaries := make([]DimensionSummary, len(dimensions))
+
+	for i, dim := range dimensions {
+		dataType := getDimensionDataType(dim)
+
+		minValue, maxValue, sampleValue, hasNulls, err := extractDimensionStats(values, i)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract stats for dimension %s: %w", dim.Name, err)
+		}
+
+		summaries[i] = DimensionSummary{
+			Name:     dim.Name,
+			DataType: dataType,
+			MinValue: minValue,
+			MaxValue: maxValue,
+			HasNulls: hasNulls,
+			Value:    sampleValue,
+		}
+	}
+
+	return &SummaryResult{
+		Dimensions: append(timeDimensionSummaries, summaries...),
+		TimeRange:  defaultTimeRange,
+	}, nil
+}
+
+func extractDimensionStats(values []interface{}, index int) (minVal, maxVal, sample interface{}, hasNulls bool, err error) {
+	baseIndex := index * statsPerDimension
+
+	minPtr, ok := values[baseIndex].(*interface{})
+	if !ok {
+		return nil, nil, nil, false, fmt.Errorf("invalid min value type")
+	}
+
+	maxPtr, ok := values[baseIndex+1].(*interface{})
+	if !ok {
+		return nil, nil, nil, false, fmt.Errorf("invalid max value type")
+	}
+
+	hasNullsPtr, ok := values[baseIndex+2].(*bool)
+	if !ok {
+		return nil, nil, nil, false, fmt.Errorf("invalid has_nulls type")
+	}
```

Comment:
Does this work? I wonder if it might always be a `*interface{}` because that's how it's created in the loop that creates `values`. The pointer may need to be dereferenced before you can cast the `interface{}` to a `bool`. Not sure though.